redaxo/src/addons/structure/lib/api_functions/api_category_move.php

```diff
@@ -8,26 +8,28 @@ class rex_api_category_move extends rex_api_function
 {
     public function execute()
     {
-        // the category to move
-        $catId = rex_request('category-id', 'int');
-        // the destination category in which the given category will be moved
-        $newCatId = rex_request('new-category-id', 'int');
-        // a optional priority for the moved category
-        $newPriority = rex_request('new-priority', 'int', 0);
+        // The category to move
+        $article_id = rex_request('article_id', 'int');
```

Comment:
Die parameter einer api function kann man nicht einfach so ändern. Es sollte rückwärtskompatibel sein.. es gibt sicherlich code der erwartet dass hier eine category-id als url parameter übergeben werden kann...

Reply:
Ich glaube, das ist hier nicht unbedingt nötig. Die api_function funktionierte in der ursprünglichen Version nicht und wurde soweit ich das nachprüfen konnte von Redaxo nicht genutzt, da das über die Content-Seite erledigt wurde. Dort wurden die jetzt eingetragenen Variablen genutzt. 
Zur Sicherheit könnte man natürlich beide Variablen abfragen.

Reply:
Diese Zeile löst immer eine Fehlermeldung aus:  `rex_category_service::editCategory($catId, 0, $data);`, denn clang wird hier auf 0 gesetzt und in der Methode nicht mehr geprüft. Daher war die api_function in ihrer ursprünglichen Form fehlerhaft.

Reply:
@gharlan  meinungen dazu? ich kann eigentlich nicht glauben dass wir hier mit dem request parameter nen bug hatten.. vllt. versteh ich es aber auch falsch

Reply:
@staabm, @gharlan Ich gehe davon aus, dass die api_function nicht benutzt und der Fehler daher nicht bemerkt wurde. Denn das Verschieben wurde bisher über die alte Funktionalität, noch aus Redaxo 4 erledigt: https://github.com/redaxo/redaxo/blob/master/redaxo/src/addons/structure/plugins/content/pages/content.php#L399-L413
Die api_function wurde innerhalb von Redaxo gar nicht aufgerufen.

Reply:
Da die alte Variante sowieso nicht funktioniert hat, und nicht verwendet wurde, passt das so denke ich.